src/MongoDB/CursorId.c

```diff
@@ -74,14 +74,22 @@ static HashTable* php_phongo_cursorid_get_properties_hash(phongo_compat_object_h
 	{
 		zval value;
 
-		if (is_debug) {
+		if (is_serialize) {
+			if (intern->id > INT32_MAX || intern->id < INT32_MIN) {
+				ZVAL_INT64_STRING(&value, intern->id);
+			} else {
+				ZVAL_LONG(&value, intern->id);
```

Comment:
Per the comment above, this will need to be removed unless `php_phongo_cursorid_init_from_hash` is improved to also accept an integer.

Since this issue wasn't caught by any new tests, I'll be sure to add a regression test before modifying either function.

Reply:
Decided to keep the old behavior of always serializing "id" as a string. Also added a regression test case to each of `cursorid-serialization-001.phpt` and `cursorid-serialization-002.phpt` for that behavior.